docs/build/js-client/10-marine-js.md

```diff
@@ -1,63 +1,138 @@
-# Marine JS
+# Marine JS (experimental)
 
-# TODO: adapt to JS Client
+> This section is experimental as the API is not stable and probably will change.
 
-## Using Marine services in Fluence JS
+## Intro
 
-Fluence JS can host Marine services with Marine JS. Currently only pure single-module services are supported.
+As you already know, Fluence carries a notion of serverless decentralized computation.
+In that ecosystem JS client has been presented more like an HTTP gateway pointed to vast Fluence network consisted of Nox relays.
 
-Before registering the service, the corresponding WASM file must be loaded. Fluence JS package exports three helper functions for that.
+Nonetheless, there are some cases when you want to host some logic locally.
+Local environment could be Node.js server or browser page.
 
-**loadWasmFromFileSystem**
+### Example 1
 
-Loads the WASM file from the file system. It accepts the path to the file and returns a buffer compatible with the `FluencePeer` API.
+Text editor application for multiple users.
+For example, 5–10 users open a web page, and they are able to simultaneously edit text, add symbols or remove words.
+Accordingly, each user has its own JS client up and running.
+JS client hosts logic (in a dedicated service or services) for interacting with incoming particles.
+Each particle carries a message of new typed symbol or added exclamation mark and more,
+and the client contains logic to process it.
 
-This function can only be used in nodejs. Trying to call it inside a browser will throw an error.
+### Example 2
 
-**loadWasmFromNpmPackage**
+AI image library with an ability to execute incoming processing requests and distribute the work through Fluence network.
+The lib should do some simple processing locally, e.g., directly in a browser and also parallel the work if necessary.
 
-Locates a WASM file in the specified npm package and loads it. The function accepts two arguments:
+### Summary
 
-- Name of an npm package
-- Path to a WASM file relative to the npm package
+These use cases have something in common—they need to host a logic locally.
+Moreover, sometimes writing a logic in pure JavaScript is not feasible.
+For example, AI processing logic would be a lot effective if written in low-level language like Rust
 
-This function can only be used in nodejs. Trying to call it inside a browser will result in an error.
+## Use cases
 
-**loadWasmFromServer**
+Usually you don't need to interact with Marine JS directly except the following:
+- Cross-platform service which works both on Nox peer and JS client peer
+- Heavy computations, CPU intensive tasks
+- Would prefer to write service in Rust or other language with compilation to WASM
 
-Loads a WASM file from the service hosting the application. It accepts the file path on server and returns a buffer compatible with the `FluencePeer` API.
+In any matched case, writing your service as WASM more suitable than doing so in JavaScript.
```

Comment:
Don't really get the meaning of this sentence, also seems like grammar is not correct or something
I can guess what you wanna say but I think this sentence can be removed and the general meaning is still there cause you have the points above and they seem to be enough

Reply:
I think there no need for summary because you have a sentence in the begining which is already kinda explains what this points are about. Anyway. `In any matched case` sounds a bit strange. Maybe `In any case` or `in any of these cases`. And you also lack `is` in `as WASM is more suitable`. But I would suggest to just to replace the sentence

`Usually you don't need to interact with Marine JS directly except the following:`

with

`Here are the reasons why you might want to write a Marine JS Wasm service instead of writing a JavaScript service for JS client`